### ISSUE TITLE
No identifiers allowed directly after numeric literal

### ISSUE THREAD

**Issue by snibo13** [realm/realm-js#6194]
### How frequently does the bug occur?

-- select --

### Description

Whenever I attempt to use useQuery from '@realm/react'. (VS Code also seems to have trouble finding this module). I can import it without any immediate issues but when I try to useQuery to retrieve values I get "No identifiers allowed directly after numeric literal"

### Stacktrace & log output

```shell
When I run this using expo, I get "no stack track" written in the log
```

### Can you reproduce the bug?

-- select --

### Reproduction Steps

Install realm/react
import useQuery from '@realm/react'
Create and render a React app with any content that has a constant equal to useQuery()

Minimal reproduction:
```
import React, { useEffect, useState } from 'react';
import { ScrollView, View, Text, TextInput, Button } from 'react-native';
import { ListItem } from 'react-native-elements';
import { StyleSheet } from "react-native";
import { useQuery } from '@realm/react'
import { createRealmContext } from '@realm/react';
import Realm, { ObjectSchema } from 'realm';
export default function App() {

  const data = useQuery();
  return (
    <View>
      <Text>hello</Text>
    </View>);

}
```

My code: [here]( https://github.com/snibo13/K/tree/with-realm)

### Version

0.4.1 and 0.4.3

### What services are you using?

-- select --

### Are you using encryption?

-- select --

### Platform OS and version(s)

React Native on Android 14, development on Windows 11

### Build environment

Which debugger for React Native: ..

**Comment by elle-j** [comment]
Hi @snibo13, from your [package.json](https://github.com/snibo13/K/blob/with-realm/package.json) I can see you're using `"realm": "^12.2.1"`. For v12.2.1 you will need to upgrade realm/react to at least `0.5.0` (e.g. our latest `"@realm/react": "^0.6.1"`). (You can refer to our [compatibility matrix](https://github.com/realm/realm-js/blob/main/COMPATIBILITY.md).)

For using realm and realm/react in your application, please see our [docs and guides](https://www.mongodb.com/docs/realm/sdk/react-native/) or [example apps](https://github.com/realm/realm-js/tree/main/examples). We recommend using realm/react's `AppProvider`, `UserProvider`, and `RealmProvider`, rather than `createRealmContext`.

**Comment by elle-j** [comment]
@snibo13, I'll go ahead and close this issue, feel free to reopen should your problem still exist.

**Comment by ravindraguptacapgemini** [comment]
@elle-j is it required to use @realm/react with realm version 12.2.1, because I am only working with realm till now?

**Comment by elle-j** [comment]
No, version 12.2.1 is not a requirement. But to use v12 and up, in combination with @realm/react, you'd need to use at least @realm/react v0.5.0. To see what enhancements and bugfixes each new version has added you can refer to our [Releases](https://github.com/realm/realm-js/releases), if you decide to use a lower version than the latest.

**Comment by ravindraguptacapgemini** [comment]
@elle-j I am getting this error with 12.2.1 without using @realm/react. How to fix this issue with realm standalone?

**Comment by takameyer** [comment]
We have noticed that this issue occurs when `hermes` is disabled on Android.  I'll open this back up so we can track and report progress.

**Comment by kraenhansen** [comment]
For future reference, the reason this is experienced only on Android might be, that on iOS the app will use the JSC which is provided by the operating system, whereas on Android it ships with a specific version provided by the React Native project. This could relate to a difference in the support for these numeric separators across different versions of JSC.

**Comment by kneth** [comment]
@snibo13 I believe we have a fix in #6289.
